Library/Homebrew/tap.rb

```diff
@@ -1011,6 +1011,17 @@ def aliases
     Homebrew::API::Formula.all_aliases.keys
   end
 
+  # @private
+  sig { returns(T::Array[Pathname]) }
+  def formula_files
+    return super if Homebrew::EnvConfig.no_install_from_api?
+
+    # This doesn't make sense for API usage.
+    odeprecated "CoreTap#formula_files"
```

Comment:
```suggestion
    odeprecated "CoreTap#formula_files (under API mode)"
```

or whatever

Reply:
Or thinking about this more: it maybe needs to be an `installed?` check instead. We do have commands that scope `HOMEBREW_NO_INSTALL_FROM_API` around the CLI parser, so the actual contents of the commands may not necessarily have `HOMEBREW_NO_INSTALL_FROM_API` set, but the tap is probably installed at that point.